pkg/controllers/management/aks/aks_cluster_handler.go

```diff
@@ -0,0 +1,426 @@
+			if conditionErr != nil {
+				return cluster, conditionErr
+			}
+		}
+		return cluster, err
+	}
+
+	// set driver name
+	if cluster.Status.Driver == "" {
+		cluster = cluster.DeepCopy()
+		cluster.Status.Driver = apimgmtv3.ClusterDriverAKS
+		return e.ClusterClient.Update(cluster)
+	}
+
+	// get aks Cluster Config, if it does not exist, create it
+	aksClusterConfigDynamic, err := e.DynamicClient.Namespace(namespace.GlobalNamespace).Get(context.TODO(), cluster.Name, v1.GetOptions{})
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return cluster, err
+		}
+
+		cluster, err = e.SetUnknown(cluster, apimgmtv3.ClusterConditionWaiting, "Waiting for API to be available")
+		if err != nil {
+			return cluster, err
+		}
+
+		aksClusterConfigDynamic, err = buildAKSCCCreateObject(cluster)
+		if err != nil {
+			return cluster, err
+		}
+
+		aksClusterConfigDynamic, err = e.DynamicClient.Namespace(namespace.GlobalNamespace).Create(context.TODO(), aksClusterConfigDynamic, v1.CreateOptions{})
+		if err != nil {
+			return cluster, err
+		}
+	}
+
+	aksClusterConfigMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&cluster.Spec.AKSConfig)
+	if err != nil {
+		return cluster, err
+	}
+
+	// check for changes between aks spec on cluster and the aks spec on the aksClusterConfig object
+	if !reflect.DeepEqual(aksClusterConfigMap, aksClusterConfigDynamic.Object["spec"]) {
+		logrus.Infof("change detected for cluster [%s], updating AKSClusterConfig", cluster.Name)
+		return e.updateAKSClusterConfig(cluster, aksClusterConfigDynamic, aksClusterConfigMap)
+	}
+
+	// get aks Cluster Config's phase
+	status, _ := aksClusterConfigDynamic.Object["status"].(map[string]interface{})
+	phase, _ := status["phase"]
+	failureMessage, _ := status["failureMessage"].(string)
+
+	switch phase {
+	case "creating":
+		if cluster.Status.GKEStatus.UpstreamSpec == nil {
+			cluster, err = e.setInitialUpstreamSpec(cluster)
+			if err != nil {
+				return cluster, err
+			}
+			return cluster, nil
+		}
+
+		e.ClusterEnqueueAfter(cluster.Name, enqueueTime)
+		if failureMessage == "" {
+			logrus.Infof("waiting for cluster AKS [%s] to finish creating", cluster.Name)
+			return e.SetUnknown(cluster, apimgmtv3.ClusterConditionProvisioned, "")
+		}
+		logrus.Infof("waiting for cluster AKS [%s] create failure to be resolved", cluster.Name)
+		return e.SetFalse(cluster, apimgmtv3.ClusterConditionProvisioned, failureMessage)
+	case "active":
+		if cluster.Spec.AKSConfig.Imported {
+			if cluster.Status.AKSStatus.UpstreamSpec == nil {
+				// non imported clusters will have already had upstream spec set
+				return e.setInitialUpstreamSpec(cluster)
+			}
+
+			if apimgmtv3.ClusterConditionPending.IsUnknown(cluster) {
+				cluster = cluster.DeepCopy()
+				apimgmtv3.ClusterConditionPending.True(cluster)
+				cluster, err = e.ClusterClient.Update(cluster)
+				if err != nil {
+					return cluster, err
+				}
```

Comment:
It is better to return here.
```suggestion
				return e.ClusterClient.Update(cluster)
```

Reply:
ok

Reply:
This still needs to be updated.

Reply:
this isn't resolved